src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala

```diff
@@ -895,8 +904,37 @@ case class InternalAzureStorage(param: MergedParameters,
             storageInfo(StorageInfo.SMK_ID)
           )
       blob.setMetadata(meta)
-      new CipherOutputStream(blob.openOutputStream(), cipher)
+
```

Comment:
Can you also log how many times we've retried?

Reply:
The retry count has been logged by the caller of this function: createDownloadStreamWithRetry() and uploadPartition() when retry is triggered.